Client/Frontend/Browser/PageZoom/PageZoomView.swift

```diff
@@ -0,0 +1,208 @@
+                             1.50, 1.75, 2.00,
+                             2.50, 3.0]
+  
+  var dismiss: (() -> Void)?
+  
+  init(webView: WKWebView?) {
+    self.webView = webView
+    
+    if let url = webView?.url,
+       let domain = Domain.getPersistedDomain(for: url) {
+      
+      _currentValue = State(initialValue: domain.zoom_level?.doubleValue ?? Preferences.General.defaultPageZoomLevel.value)
+    } else {
+      _currentValue = State(initialValue: webView?.value(forKey: PageZoomView.propertyName) as? Double ?? Preferences.General.defaultPageZoomLevel.value)
+    }
+  }
+  
+  var body: some View {
+    VStack(spacing: 0.0) {
+      Divider()
+      HStack(spacing: 0.0) {
+        Text(Strings.PageZoom.zoomViewText)
+          .font(.system(.subheadline))
+          .frame(maxWidth: .infinity, alignment: .leading)
+        
+        Spacer()
+        
+        ZoomView(
+          minValue: minValue,
+          maxValue: maxValue,
+          value: $currentValue,
+          onDecrement: decrement,
+          onReset: reset,
+          onIncrement: increment)
+          .frame(maxWidth: .infinity)
+        
+        Spacer()
+        Button {
+          dismiss?()
+        } label: {
+          Image(systemName: "xmark")
+            .renderingMode(.template)
+            .foregroundColor(Color(UIColor.braveBlurple))
+            .font(.system(.footnote).weight(.medium))
+        }
+        .frame(maxWidth: .infinity, alignment: .trailing)
+      }
+      .padding(.horizontal)
+      .padding(.vertical, 7.0)
+    }
+    .background(Color(UIColor.braveBackground))
+  }
+  
+  private func storeChanges() {
+    guard let webView = webView,
+          let url = webView.url else { return }
+    
+    webView.setValue(currentValue, forKey: PageZoomView.propertyName)
+    
+    let domain = Domain.getPersistedDomain(for: url)?.then {
+      $0.zoom_level = NSNumber(value: currentValue)
+    }
+    
+    try? domain?.managedObjectContext?.save()
+  }
+  
+  private func increment() {
+    guard let index = PageZoomView.steps.firstIndex(of: currentValue) else { return }
+    currentValue = PageZoomView.steps[index + 1]
+    storeChanges()
+  }
+  
+  private func reset() {
+    currentValue = 1.0
+    storeChanges()
+  }
+  
+  private func decrement() {
+    guard let index = PageZoomView.steps.firstIndex(of: currentValue) else { return }
+    currentValue = PageZoomView.steps[index - 1]
+    storeChanges()
+  }
+}
+
+#if DEBUG
+struct PageZoomView_Previews: PreviewProvider {
+  static var previews: some View {
+    PageZoomView(webView: nil)
+      .previewLayout(PreviewLayout.sizeThatFits)
+  }
+}
+#endif
+
+class PageZoomViewController: UIHostingController<PageZoomView> {
+
+  init(webView: WKWebView) {
+    super.init(rootView: PageZoomView(webView: webView))
+  }
+
+  @objc required dynamic init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+}
```

Comment:
TBH this is probably not needed at all, no additional changes aside from a name change so you are better off just using `UIHostingController` directly at init point.